graphics/partio/Portfile

```diff
@@ -30,3 +32,24 @@ depends_build-append \
                     port:doxygen \
                     port:swig-python
 depends_lib         port:zlib
+
+post-destroot {
+    if {![file exists ${destroot}${prefix}/share/${name}/py]} {
+        xinstall -d ${destroot}${prefix}/share/${name}/py
+    }
+    copy ${worksrcpath}/src/py/example \
+         ${destroot}${prefix}/share/${name}/py
+}
+
+variant tests description {Build tests} {
+    depends_lib-append      port:gtest
+    configure.args-append   GTEST_ENABLED=ON
+    test.run    yes
+
+    post-destroot {
+        copy ${worksrcpath}/src/data \
+             ${destroot}${prefix}/share/${name}/test
+    }
+}
+
+default_variants    +tests
```

Comment:
If I'm not mistaken, `partio` is available to users as a [binary archive](https://packages.macports.org/partio/), so users shouldn't need to build the tests themselves.

In addition, according to the [MacPorts Guide](https://guide.macports.org/), in section 4.7.3, it says to ["Install Docs and Examples"](https://guide.macports.org/#development.practices.install-docs). Despite the rather disconcerting "TODO" label for the section, I believe that it is considered best practice to make it so that the default variant installs all documentation, tutorials, example files, etc. in order to produce a "complete package". Since the `partio` build files include the compiled the unit tests themselves as output files, I have put them into the same category as the rest of the documentary bits.

Reply:
> partio is available to users as a binary archive, so users shouldn't need to build the tests themselves.

That's true. But if we can help those that are building from source for whatever reason, whilst also reducing the complexity of the build process, I think that's no bad thing. As an example, we wouldn't make the build process do things unnecessarily on the premise that the user will never see it.

> Since the partio build files include the compiled the unit tests themselves as output files, I have put them into the same category as the rest of the documentary bits

Aren't the configure flags in the test variant what leads to the unit tests being compiled?

Either way, I don't feel too strongly about this, and you have very kindly chosen to maintain this port (thank you 😄). Whether these tests count as documentation/tutorials/etc. or not is up to interpretation, and if you think this is what's best for the port, I'm more than happy to go with you on this.

Reply:
sometimes (most times) variants propagate through to all the dependencies. So if you have "port XYZ +tests" then every other thing will be built with the "+tests" variant as well. And unfortunately lots of things have a variant named "+tests" so that could be a lot of stuff.

I tried to get an idea started to have a custom variant for every port called "${name}-tests" to get around this mess of unrequested propagated deps, but that got exactly ZERO purchase, and everyone thought it was a horrible idea.

So -- default variants named "+test" are bad news, possibly.

Reply:
Upon further reflection, I've come to the conclusion that +tests should not be in the default variants for this particular port. Since `partio`'s unit tests rely on `gtest`, I agree that there's no reason why a user should be forced to install `gtest` in order to have a working `partio` default package. I will change to commit to remove the default variant.

Responding to @kencu's comment:

> sometimes (most times) variants propagate through to all the dependencies. So if you have "port XYZ +tests" then every other thing will be built with the "+tests" variant as well. And unfortunately lots of things have a variant named "+tests" so that could be a lot of stuff.

Although there are some package management systems that behave this way, I was under the impression that the only way to get MacPorts to behave in this cascading fashion was to put a `require_active_variants <deppkg> tests` into the top-level portfile. I've already encountered instances where a portfile that I wrote that has a `+python38` variant won't automatically install all of its dependencies that have variants `+python38`. It's possible that the reason why it didn't work on my machine was that those dependencies were already installed with `+python37`, but still, it's an example of a case where the variant _did not_ propagate into the dependencies as I would have expected. (I would have expected the `<deppkg> +python37` to get deactivated and the `<deppkg> +python38` variant to get installed.)

Reply:
yes, if the port is already installed, then it would not be re-examined for it's variants.

another thing that some genius with a lot of time on his/her hands might dig on -- I have no idea how to fix that, without making a total mess of things...